### ISSUE TITLE
UnboundLocalError: local variable 'wandb_logger' referenced before assignment

### ISSUE THREAD

**Issue by wq9** [ultralytics/yolov5#2562]
## 🐛 Bug
Multi-GPU Training does not work with latest commit (https://github.com/ultralytics/yolov5/commit/e8fc97aa3891f05812d7dfff90ca66d3481bda2c Improved W&B integration) 

## To Reproduce (REQUIRED)
```
python -m torch.distributed.launch --nproc_per_node 4 train.py --data coco128.yaml --cfg yolov5s.yaml --weights '' --batch-size 16
```

Output:
```
Traceback (most recent call last):
  File "train.py", line 537, in <module>
Traceback (most recent call last):
  File "train.py", line 537, in <module>
    train(hyp, opt, device, tb_writer)
  File "train.py", line 76, in train
    train(hyp, opt, device, tb_writer)
  File "train.py", line 76, in train
    loggers = {'wandb': wandb_logger.wandb}  # loggers dict
UnboundLocalError: local variable 'wandb_logger' referenced before assignment
    loggers = {'wandb': wandb_logger.wandb}  # loggers dict
UnboundLocalError: local variable 'wandb_logger' referenced before assignment
Traceback (most recent call last):
  File "train.py", line 537, in <module>
    train(hyp, opt, device, tb_writer)
  File "train.py", line 76, in train
    loggers = {'wandb': wandb_logger.wandb}  # loggers dict
UnboundLocalError: local variable 'wandb_logger' referenced before assignment
```

## Expected behavior
Normal Multi-GPU Training

## Environment
N/A

## Additional context
`wandb_logger` is not defined for child processes, so child processes terminates prematurely (https://github.com/ultralytics/yolov5/commit/e8fc97aa3891f05812d7dfff90ca66d3481bda2c#diff-ed183d67207df065a11e1289f19d34cc2abbc5448dea952683cfe9728c342b95R76)

**Comment by glenn-jocher** [comment]
@wq9 thank you for the bug report!

@AyushExel do you think there's a quick fix for this (i.e. next few hours)? If not we should probably revert https://github.com/ultralytics/yolov5/pull/2125 until we have a chance to debug further.

**Comment by AyushExel** [comment]
@glenn-jocher Yes I'm looking into this now. Any idea what's the difference  in execution in multi-GPU case?

**Comment by AyushExel** [comment]
@glenn-jocher Okay I think I found the solution. Testing it in a bit. I'll let you know

**Comment by glenn-jocher** [comment]
@AyushExel I'm not sure what it could be. Multi-GPU is always tricky. Rank 0 and -1 processes are intended for the main process/GPU, while all other parts of the code will run on all processes.

Do you have access to a Multi-GPU instance to test on?

**Comment by AyushExel** [comment]
@glenn-jocher  Yes, I'm testing on multi-GPUS. I usually work on single GPU so this test didn't cross my mind. In the previous integration, wandb logging worked only with the single GPU case right? Juding by the previous integration, here:
```
if rank in [-1, 0] and wandb and wandb.run is None:
```

**Comment by glenn-jocher** [comment]
@AyushExel oh, no, master before PR performed W&B logging with Multi-GPU well, i.e.:
<img width="1014" alt="Screenshot 2021-03-23 at 15 25 45" src="https://user-images.githubusercontent.com/26833433/112162178-1e30f600-8bec-11eb-87f7-1abf292c9791.png">

See [Multi-GPU Tutorial](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training) for DP and DDP commands (we need to test both). DDP is strongly preferred though due to it's better performance.

**Comment by AyushExel** [comment]
@glenn-jocher thanks for the info. I'm working on a fix.

**Comment by AyushExel** [comment]
@glenn-jocher Okay this is fixed. I'm testing in a different environment without wandb installed. Expect a PR in a few minutes

**Comment by AyushExel** [comment]
@glenn-jocher I've tested on my end and the fix seems to work in all cases. CI tests are running on the PR now. Hopefully, I haven't introduced any CI typos. 
Sorry about this bug though. I'll include DDP in my test cases.

**Comment by glenn-jocher** [comment]
@wq9 thanks for identifying this problem! We've pushed a fix for this now in PR #2574, and made sure to test Multi-GPU training as well as resuming Multi-GPU training. Please git pull or clone again to receive this update, and let us know if you have any further problems!
